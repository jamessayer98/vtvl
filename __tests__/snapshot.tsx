import { render, screen, fireEvent } from '@testing-library/react';
import Home from '../pages/index';

it('renders homepage unchanged', () => {
  const { container } = render(<Home />);
  expect(container).toMatchSnapshot('homepage');
});

it('renders Import CSV modal unchanged', () => {
  const { container } = render(<Home />);

  const button = screen.getByText('Import');
  fireEvent.click(button);

  expect(container).toMatchSnapshot('import csv');
});
