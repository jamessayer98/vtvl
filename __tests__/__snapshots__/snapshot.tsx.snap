// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Import CSV modal unchanged: import csv 1`] = `
<div>
  <div
    class="flex min-h-screen flex-col items-center justify-center py-2"
  >
    <main
      class="flex w-full flex-1 flex-col items-center justify-center px-20 text-center"
    >
      <button
        class="px-6 py-3 rounded-full border text-gray-700 w-96 hover:bg-primary hover:border-primary hover:text-white"
      >
        Import
      </button>
      <div
        class="absolute w-full h-full flex justify-center items-center bg-[#a8a8a8] shadow-xl"
      >
        <div
          class="rounded-3xl p-6 bg-white w-fit min-w-[480px]"
        >
          <div
            class="flex justify-between"
          >
            <div
              class="flex items-center"
            >
              <span
                class="font-medium text-lg leading-7 mr-3"
              >
                Import from CSV file
              </span>
              <span
                class="font-bold text-xs leading-6 px-2 rounded-full bg-[#EFF2F5] text-[#58667E]"
              >
                1
                 of 
                3
              </span>
            </div>
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2712%27%20height=%2712%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="close"
                class="cursor-pointer"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="text-sm text-gray-500 mt-2 text-start"
          >
            Upload and create user to this schedule.
          </div>
          <div
            class="flex flex-col justify-center rounded-lg border border-gray-200 mt-5 py-4 px-6"
            data-testid="drop-target"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <div
              class="text-sm mt-3"
            >
              <span
                class="text-[#F9623B] font-medium cursor-pointer"
              >
                Click to upload
              </span>
               
              or drag and drop
            </div>
            <div
              class="text-xs leading-[18px]"
            >
              CSV (max. size 1mb)
            </div>
          </div>
          <div
            class="my-8 text-start"
          >
            Download:
             
            <a
              class="font-bold text-[#1A73E8] cursor-pointer"
              download="template.csv"
            >
              VTVL User Template
            </a>
          </div>
          <div
            class="flex gap-3"
          >
            <button
              class="border border-gray-300 shadow-xs rounded-full py-[10px] px-[18px] flex-1"
            >
              Cancel
            </button>
            <button
              class="bg-primary border border-primary text-white shadow-xs rounded-full py-[10px] px-[18px] flex-1"
              disabled=""
            >
              Upload file
            </button>
            <input
              accept="text/csv, .csv"
              data-testid="file-upload"
              hidden=""
              id="file-upload"
              type="file"
            />
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`renders homepage unchanged: homepage 1`] = `
<div>
  <div
    class="flex min-h-screen flex-col items-center justify-center py-2"
  >
    <main
      class="flex w-full flex-1 flex-col items-center justify-center px-20 text-center"
    >
      <button
        class="px-6 py-3 rounded-full border text-gray-700 w-96 hover:bg-primary hover:border-primary hover:text-white"
      >
        Import
      </button>
    </main>
  </div>
</div>
`;
